### PR TITLE
[js/web] support multi-thread for wasm backend

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: support multi-thread WebAssembly in onnxruntime web.
- uses global thread-pool instead of per session thread pool by default.
- by default `numThreads` is set to [this value](https://github.com/microsoft/onnxruntime/compare/fs-eire/ort-web-support-threads?expand=1#diff-2f4348a4ffcae301654c5699f5d8045eb2280660c74d1894657c028201a1346aR20-R23)
- disable '-flto' for multi-thread wasm build.

This PR also comes with several changes and optimizations:
- move flag declarations into `onnxruntime-common`
- onnxruntime-web test runner no longer depends on wasm artifacts if specified with "-b=webgl"
- optimized a few rules of ESLint
- increase test timeout when in debug mode
- a few code cleanups

### CODE REVIEW

File: onnxruntime/wasm/api.cc [hunk 1]
```diff
@@ -12,17 +12,25 @@ namespace {
 Ort::Env* g_env;
 }  // namespace
 
-void OrtInit() {
-  // TODO: allow user to specify logging
-  OrtLoggingLevel logging_level = ORT_LOGGING_LEVEL_WARNING;
-  g_env = new Ort::Env{logging_level, "Default"};
+void OrtInit(int numThreads, int logging_level) {
```
Comment (hanbitmyths) [inline onnxruntime/wasm/api.cc:15]: is there a warning for unused parameter?

**Comment by hanbitmyths** [comment]
WebGL team can approve if there is no concern. I'm okay with changes.

File: js/web/lib/wasm/session-handler.ts [hunk 2]
```diff
@@ -113,7 +113,7 @@ export class OnnxruntimeWebAssemblySessionHandler implements SessionHandler {
   loadModel(model: Uint8Array): void {
     const wasm = getInstance();
     if (!ortInit) {
-      wasm._OrtInit();
+      wasm._OrtInit(env.wasm.numThreads!, 2 /* LoggingLevel::Warning */);
```
Comment (hanbitmyths) [inline js/web/lib/wasm/session-handler.ts:116]: Add a logging level into Flags?

Comment (fs-eire) [reply js/web/lib/wasm/session-handler.ts:116]: will be separated PR together with SessionOptions/RunOptions support

**Review by hanbitmyths — APPROVED** [review]

**Pull request merged**